File: src/Block/EndWhileBlock.java
/**
 * This file copyright (c) 2015-2016, Jing Fan, Adalbert Gerald Soosai Raj, and Jignesh M. Patel
 *
 * See the file CREDITS.txt in the root directory for details.
 **/

package Block;

/**
 * @brief The SQL block for end while.
 */
public class EndWhileBlock extends Block {
  private String endStr;

  /**
   * @brief Constructor.
   * @param stage The string indicates the stage of this code block.
   * @param indent The indent level.
   * @param endStr The string indicates the termination condition of the
   *               iterations.
   * @param counter The name of the table that is used to control the
   *                 iterations.
   */
  public EndWhileBlock(String stage,
                       int indent,
                       String endStr,
                       String counter) {
    super(stage, indent);
    this.endStr  = endStr;
    if (endStr.equals("NO_MESSAGE")) {
      this.append("SELECT @flag = COUNT (*) FROM " + counter + ";");
    } else {
      this.append("SET @flag = @flag - 1");
    }
    this.append("END", indent - 1);
    this.sql = this.sb.toString();
  }

  /**
   * @brief Get the termination condition.
   * @return The termination condition.
   */
  public String getEndStr() {
    return endStr;
  }
}

File: python/src/Blocks/EndWhileBlock.py
from Block import Block
class EndWhileBlock(Block):
    '''
    generates sql to end a while loop
    '''
    
    endStr=""
    
    def getEndStr(self):
        '''
        returns the ending string
        '''
        return self.endStr
    def __init__(self,stage,indent,endStr,counter):
        '''
        constructor: generates the sql for end while
        @param stage: stage of the block
        @type stage: string
        @param indent: indent level
        @type indent: int
        @param endStr: end string 
        @type endStr: string
        @param counter: table name thats used to decide the number of iterations
        @type counter: string
        '''
        super(EndWhileBlock,self).__init__(stage,indent)
        self.endStr=endStr
        if(self.endStr=="NO_MESSAGE"): 
            self.append("flag := (SELECT COUNT (*) FROM " + str(counter) + ");")
        else:
            self.append("flag := flag - 1;") 
        self.append("END LOOP;",indent-1)
        self.append("\nEND$$")
        self.sql=self.sb            